index.html

```diff
@@ -1145,9 +1230,15 @@ <h2>Verification Methods</h2>
 The `controller` property is used by [=controller documents=], as described in
 Section [[[#controller-documents]]], and by [=verification methods=], as
 described in Section [[[#verification-methods]]]. When it is used in either
-place, its purpose is the same; that is, it expresses one or more entities that
-are authorized to perform certain actions associated with the resource with
-which it is associated. To ensure explicit security guarantees, the
+place, its purpose is essentially the same; that is, it expresses one or more 
+entities that are authorized to perform certain actions associated with the 
+resource with which it is associated. 
+
+In the case of the controller of a controller document, the controller can 
+update the content of the document. In the case of the controller of a 
+verification method, the controller can generate proofs that satisfy the method.
```

Comment:
You are correct.

This is the conflation/confusion with the controller property.

The controller of a controller document an update anything in the document. It doesn't not mean they can generate any particular proofs (but one would expect the controller to update the verification methods such that they can).

The controller of a verification method can create proofs that satisfy the method. It does not mean they can update the verification method.

To me, this makes these two terms fundamentally different, but that's a big normative change.

Reply:
I think it would be wise to have different terms for the controller of a controller document and the controller of a verification method. e.g. document controller and proof controller. In this way the reader can see that they are two different roles.

Reply:
I have tried to introduce these two different terms/roles to make the semantics clearer. But I have not caught all the changes that are necessary because this PR does not include all instances of the controller term that need to be replaced by the term 'document controller'.

Reply:
@David-Chadwick Before we go in that direction, I'd like to see if we have buy in from enough implementers to change the term definitions.

Reply:
I fundamentally agree with @David-Chadwick that we seem to conflate two different notions into one, which creates more problems than it solves. Ie, technically speaking, I would also prefer to separate them.

But @jandrieu may be right: it is unclear what this means for deployed implementation of DIDs and VCs.

---

Note that we can play an ugly trick. We can modify the JSON-LD `@context` so that the two uses of the`controller` term map to different URLs. That is what "scoped contexts" are for (see an [Example in the JSON-LD spec](https://www.w3.org/TR/json-ld11/#example-46-defining-an-context-within-a-term-definition-used-on-type)). In plain text, this means the two usage of the same terms are formally defined as separate URLs and the rest is done behind the scenes doors. This may still create problems for those who rely on the purely RDF version of the terms, though it may be in the minority. But I find it an ugly hack nevertheless. 

cc @msporny @dlongley

Reply:
@jandrieu Happy to see if we can get buy-in. From my perspective, having one term that includes two different roles is confusing, so separating the terms should help new implementors, who in the long run, are sure to outnumber existing implementors.

Reply:
I'm definitely -1 to changing the data model in this way. The `controller` property is applicable to the entity/subject on which it appears (like any other property). We don't need multiple names for this (and shouldn't do it generally, e.g., we don't make a new `id` or `type` property for every possible subject on which it could appear, we just reuse them). Furthermore, this change would affect many long-lived implementations and create conflict with DID core.

However, I'm not opposed to more informative text that we believe can be helpful in this area.

Reply:
I should also say that I think that whatever subtlety is trying to be addressed here by potentially splitting the property names would not be well understood by most developers -- and I would expect many of them to just question why the same name wasn't used, as they can mentally ignore the prefix before 'controller' given the context of where the property appears.

@ivan,

> This sounds like the controller of a verification method can only generate proofs...

No, you don't know "who" or "what" (or "how many") can generate the proofs, you only know that the verification method "lives" in a controller document (which controls the verification method), and that the controller of that document has authorized such proofs for a specific purpose (through the verification relationship).

> ...but cannot change the verification method, e.g., change the expiration method or revoke the key. I would not think that is correct, is it?

Control "flows" in one direction and ultimately stems from some root of authority. Any party further up the chain has at least as much power as those down chain.

As stated above, a verification method "lives" (or is canonically/natively expressed) in a controller document and the verification method's controller value MUST match the controller document's identifier (see below for the reference), i.e., it is controlled by its controller document -- and, therefore, whomever controls the controller document always controls it. Whomever controls the verifiable data registry for the controller documents ultimately controls what gets expressed in that controller document (including its `controller` field, if present).

https://w3c.github.io/controller-document/#retrieve-verification-method

> 10) If the absolute URL value of verificationMethod.controller does not equal controllerDocumentUrl, an error MUST be raised and SHOULD convey an error type of [INVALID_VERIFICATION_METHOD](https://w3c.github.io/controller-document/#INVALID_VERIFICATION_METHOD).

Reply:
Typos in 3rd paragraph of @jandrieu's https://github.com/w3c/controller-document/pull/116#discussion_r1838335275.  It should read (emphasis mine) —
> The controller of a controller document **_can_** update anything in the document. It **_does not_** mean they can generate any particular proofs (but one would expect the controller to update the verification methods such that they can).

In conversations like this one, that go deep into the weeds, it's vital that such typos be avoided as much as possible, and corrected as quickly as possible when they do arise!

Reply:
-1 for making the distinction between `documentController` and `proofController` at this point in time -- for the reasons here: https://github.com/w3c/controller-document/pull/116/files#r1839014775